web/src/features/time/TimeController.tsx

```diff
@@ -24,13 +24,13 @@ export default function TimeController({ className }: { className?: string }) {
   );
 
   useEffect(() => {
-    if (datetimes) {
+    if (datetimes && datetimes.at(-1)) {
       // This value is stored in state to avoid flickering when switching between time averages
       // as this effect means index will be one render behind if using datetimes directly
       setNumberOfEntries(datetimes.length - 1);
       // Reset the selected datetime when data changes
       setSelectedDatetime({
-        datetimeString: dateToDatetimeString(datetimes[datetimes.length - 1]),
+        datetimeString: dateToDatetimeString(datetimes.at(-1) as Date),
```

Comment:
the behavior of .at is kind of surprising to me. like why is newWay `date | undefined` but oldWay is `date` ?  

<img width="527" alt="image" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/20405975/902b403e-8f52-4b57-b017-3a036c26b86a">
<img width="534" alt="image" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/20405975/f11aacbd-c6ba-4dab-abec-59f5e1d90b77">

Reply:
That looks like an error because they both evaluate to undefined if called on a zero length array.

As far as I know it should be exactly the same as calling `length - 1` so please do use `.at()`

See this for reference:
<img width="840" alt="image" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/30777521/3d3928a0-7ecf-4b59-8631-1192adfd8c0c">